pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -196,6 +198,51 @@ public void setupHelixEntries(final String topicName, final String realtimeTable
         nReplicas);
   }
 
+  // Remove all trace of LLC for this table.
+  public void cleanupLLC(final String realtimeTableName) {
+    // Start by removing the kafka partition assigment znode. This will prevent any new segments being created.
+    ZKMetadataProvider.removeKafkaPartitionAssignmentFromPropertyStore(_propertyStore, realtimeTableName);
+    LOGGER.info("Removed Kafka partition assignemnt (if any) record for {}", realtimeTableName);
+    // If there are any completions in the pipeline we let them commit.
+    Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
```

Comment:
I thought of thread.yield() but was not sure that I will get this thread to go after two zk operations (which may block the committing thread). So, I came up with 1s. This should be sufficient if there is no GC, but yes, if there is GC, then it will not be.

Instead of increasing this timeout I suggest we make it an operational check and re-apply the same change in case new LLC segments have come up.

I am open to other ideas as well (I also thought of 1s sleep followed by thread.yield())